requirements.in

```diff
@@ -0,0 +1,15 @@
+dune-client
+moralis
+pandas
```

Comment:
At the moment this actually removes the version pinning for these packages:
```
requests==2.31.0
web3==7.0.0
pandas==2.2.1
SQLAlchemy==2.0.28
psycopg2==2.9.9
python-dotenv==1.0.0
```
If we want to keep them pinned we should add that here. For other packages I think it might be worth pinning major versions as well just because it can't hurt. So for example we could set something like `dune-client>=1.7.5,<2.0`.

Reply:
Are any of those restrictions necessary at the moment?

I would have expected that one uses as few restrictions as possible in `requirements.in`. Any recompiling will require some form of testing of the new dependencies anyways. At that stage one can add restrictions or update the code.

Restricting on major versions sounds OK. I would probably only do it after we see an issue with some version.

Reply:
@bram-vdberg I actually see all these packages you listed fully pinned down in this PR, which i guess is the expected behavior as they are listed in the requirements.in file.

Reply:
@harisang I meant that they're pinned in the requirements.txt file, but not in the requirements.in file. This means that recompiling the requirements.in could upgrade these packages. Another way to do this is to pin selected packages in the requirements.in, that means that we can recompile and upgrade downstream dependencies without upgrading these packages.

I think Felix is right that we don't need to do that unless necessary.